ngraph/core/src/op/roi_align.cpp

```diff
@@ -190,4 +202,104 @@ namespace ngraph
     {
         return s << as_string(type);
     }
+} // namespace ngraph
+namespace
+{
+    bool evaluate_roi_align(const HostTensorVector& args,
+                            const HostTensorPtr& out,
+                            const int pooled_height,
+                            const int pooled_width,
+                            const int sampling_ratio,
+                            const float spatial_scale,
+                            const op::v3::ROIAlign::PoolingMode& pooling_mode)
+    {
+        auto feature_maps = args[0];
+        auto rois = args[1];
+        auto batch_indices = args[2];
+
+        std::vector<int64_t> batch_indices_vec_scaled_up =
+            host_tensor_2_vector<int64_t>(batch_indices);
+
+        switch (feature_maps->get_element_type())
+        {
+        case element::Type_t::bf16:
```

Comment:
Optinal: You can use `TYPE_CASE` macro in order to reduce number of lines. Look for example to https://github.com/openvinotoolkit/openvino/blob/cd722d72dfe334bc426e06e5cdf81f43841ecaf3/ngraph/core/src/op/multiply.cpp#L71

Reply:
Yeah, I really don't like this macro